torch/types.py

```diff
@@ -70,8 +70,10 @@
 
 # Meta-type for "device-like" things.  Not to be confused with 'device' (a
 # literal device object).  This nomenclature is consistent with PythonArgParser.
+ExplicitDevice: TypeAlias = Union[_device, str, int]
 # None means use the default device (typically CPU)
-Device: TypeAlias = Union[_device, str, int, None]
+OptionalDevice: TypeAlias = Union[ExplicitDevice, None]
+Device: TypeAlias = OptionalDevice
```

Comment:
Device is not optional usually !

Reply:
I can't think of a better type name.. Any advice?

Reply:
Isn't the current "Device" type used heavily today? I would not expect we can change it !

Reply:
`DeviceLike` makes less sense because str and int are treated as device in device APIs for many years.